core/src/main/java/de/javakaffee/web/msm/MemcachedElasticConnectionFactory.java

```diff
@@ -0,0 +1,108 @@
+package de.javakaffee.web.msm;
+
+import net.spy.memcached.BinaryConnectionFactory;
+import net.spy.memcached.ConnectionFactory;
+import net.spy.memcached.DefaultConnectionFactory;
+
+import java.lang.reflect.*;
+
+/**
+ * Created by asodhi on 11/29/2016.
+ */
+public class MemcachedElasticConnectionFactory implements StorageClientFactory.MemcachedNodeURLBasedConnectionFactory {
+
+    @Override
+    public ConnectionFactory createBinaryConnectionFactory(final long operationTimeout,
+                                                           final long maxReconnectDelay,
+                                                           boolean isClientDynamicMode) {
+        try {
+            Class clientMode = Class.forName("net.spy.memcached.ClientMode");
```

Comment:
You could also add the elasticache-java-cluster-client maven dependency, so that not that much reflection is needed. We already have dependencies for several client libs...